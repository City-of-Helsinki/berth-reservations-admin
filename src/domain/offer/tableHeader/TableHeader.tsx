import React from 'react';
import classNames from 'classnames';
import { useTranslation } from 'react-i18next';

import styles from './tableHeader.module.scss';

interface Tab {
  label: string;
  value: string;
  enabled: boolean;
}
export interface TableHeaderProps {
  activeFilters?: string[];
  filters: Tab[];
  filterPrefix?: string;
  handleSetFilter(filter?: string): void;
}

const TableHeader: React.SFC<TableHeaderProps> = ({
  activeFilters, // Note: this filters might include other active filters from the table
  filters,
  filterPrefix = '',
  handleSetFilter,
}) => {
  const { t } = useTranslation();

  const filtersButtons = filters.map((filter, i) => (
    <button
      key={i}
      className={classNames(styles.filterBtn, {
        [styles.active]: activeFilters?.includes(filter.label),
        [styles.disabled]: !filter.enabled,
      })}
      disabled={!filter.enabled}
      onClick={() => handleSetFilter(filter.label)}
    >
      {`${filterPrefix} ${filter.label}`}
    </button>
  ));

  const noActiveFilter =
    filters.filter(filterVal => activeFilters?.includes(filterVal.label))
      .length === 0;

  return (
    <>
      <button
        className={classNames(styles.filterBtn, {
          [styles.active]: noActiveFilter,
        })}
        onClick={() => handleSetFilter()}
      >
        {t('common.table.all')}
      </button>
      {filtersButtons}
    </>
  );
};

export default TableHeader;
