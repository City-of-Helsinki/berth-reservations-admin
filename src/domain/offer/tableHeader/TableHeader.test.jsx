import React from 'react';
import { shallow } from 'enzyme';

import TableHeader from './TableHeader';

describe('TableHeader', () => {
  const handleSetFilter = jest.fn();
  const initialProps = { filters: ['A', 'B'], handleSetFilter };

  const getWrapper = (props = {}) =>
    shallow(<TableHeader {...initialProps} {...props} />);

  beforeEach(() => {
    handleSetFilter.mockRestore();
  });

  it('renders normally', () => {
    const wrapper = getWrapper();

    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should call handleSetFilters with the filter's value when a filter button is clicked", () => {
    const firstFilter = initialProps.filters[0];
    const filterBtn = getWrapper().find(`button[children="${firstFilter}"]`);

    filterBtn.simulate('click');

    expect(handleSetFilter).toHaveBeenNthCalledWith(1, firstFilter);
  });

  it('should call handleSetFilters with undefined when the first button is clicked', () => {
    const firstBtn = getWrapper()
      .find('button')
      .first();

    firstBtn.simulate('click');

    expect(handleSetFilter).toHaveBeenNthCalledWith(1);
  });
});
