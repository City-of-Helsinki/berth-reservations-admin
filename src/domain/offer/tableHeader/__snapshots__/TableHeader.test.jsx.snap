// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableHeader renders normally 1`] = `"<button class=\\"filterBtn\\">Kaikki</button><button class=\\"filterBtn\\">A</button><button class=\\"filterBtn\\">B</button>"`;
